test/test_reads.py

```diff
@@ -263,6 +263,49 @@ def testWalkHSPLeftAndRightOverhangingMatch(self):
                           (13, 'T', False)],
                          list(read.walkHSP(hsp)))
 
+    def testcheckAlphabetwithReadMustBePermissive(self):
+        """
+        The checkAlphabet function must be permissive if a dark.Read is
+        passed.
+        """
+        read = Read('id', 'ARSTGATGCASASASASASAS')
+        self.assertEqual(set('ACGSRT'), read.checkAlphabet())
+
+    def testcheckAlphabetAAReadMatchingReturnTrue(self):
+        """
+        If an AA read with an AARead readClass is passed in, the checkAlphabet
+        function must return True.
+        """
+        read = AARead('id', 'ARSTGATGCASASASASASAS')
+        self.assertEqual(set('ACGSRT'), read.checkAlphabet())
+
+    def testcheckAlphabetDNAReadMatchingReturnTrue(self):
+        """
+        If a DNA read with a DNARead readClass is passed in, the checkAlphabet
+        function must return True.
+        """
+        read = DNARead('id', 'AAATTAACGGGCCTAGG')
+        self.assertEqual(set('ACTG'), read.checkAlphabet())
+
+    def testcheckAlphabetAAReadNotMatchingRaise(self):
+        """
+        If an AA read with a DNARead readClass is passed in, the checkAlphabet
+        function must raise an IndexError.
+        """
+        read = AARead('id', 'AAATTAACGGGCCTAGG')
+        error = ("It looks like a DNA sequence has been passed to AARead().")
```

Comment:
You don't need the parens here, and because the error is a regex, you should escape the `().` chars.